cpp/src/fil/infer.cu

```diff
@@ -356,6 +527,15 @@ void infer(storage_type forest, predict_params params, cudaStream_t stream) {
       infer_k_launcher<FLOAT_UNARY_BINARY, storage_type>(forest, params,
                                                          stream);
       break;
+    case TREE_PER_CLASS:
+      if (params.num_classes > FIL_TPB) {
+        params.leaf_algo = TREE_PER_CLASS_MANY_CLASSES;
```

Comment:
`infer_k_launcher` can set `params.leaf_algo`.

Reply:
that would gather the FIL_TPB usage together, but require the same if-else-else statement in 8 places, 4 for smem determination and 4 for actual launch. This will not even be solved by assigning function template specializations to function pointers, as the type dispatch will need to happen in a nested fasion anyway.
This will significantly inflate the code. Is that worth it?

Reply:
I could instead make blockdim_x be determined in `infer`, but if in the future, blockDim.x and gridDim.x ever get coupled, that would make it awkward.

Reply:
That is OK.